tests/rptest/tests/datalake/datalake_e2e_test.py

```diff
@@ -126,6 +126,20 @@ def generate_record(self, t):
 """
 
 AVRO_SCHEMA_TEST_CASES = {
+    "primitive":
+    AvroSchema(
+        schema_str="""{"type": "long", "name": "a_number"}""",
+        record_generator=lambda t: int(t),
+        expected_trino=
+        [('redpanda',
+          'row(partition integer, offset bigint, timestamp timestamp(6), headers array(row(key varbinary, value varbinary)), key varbinary)',
+          '', ''), ('root', 'bigint', '', '')],
+        expected_spark=
+        [('redpanda',
+          'struct<partition:int,offset:bigint,timestamp:timestamp_ntz,headers:array<struct<key:binary,value:binary>>,key:binary>',
+          None), ('root', 'bigint', None), ('', '', ''),
+         ('# Partitioning', '', ''),
```

Comment:
These expected schemas don't seem right, given they don't include a_number?

Reply:
See the commit message please. It doesn't seem that naming is supported there. At least not according to the C++ library.

Reply:
I left some name in there as an example that it is ignored. Can drop if confusing.

Reply:
Ah I conflated typename and field name in looking at this. That's interesting (and also not surprising given how gnarly the Avro node code is...).

I guess/hope that this is an uncommon case -- presumably people care about storing more than one field... Nice find regardless!